File: backend/schemas/analytics.py
from pydantic import BaseModel
from datetime import datetime


class UsageStatsResponse(BaseModel):
    total_generations: int
    total_training_hours: float
    total_storage_mb: float
    generations_this_month: int
    voices_this_month: int


class UsageTimeSeriesPoint(BaseModel):
    date: datetime
    count: int
    type: str


class UsageTimeSeriesResponse(BaseModel):
    data: list[UsageTimeSeriesPoint]
    period: str

File: frontend/src/types/analytics.ts
export interface UsageStats {
  total_generations: number;
  total_training_hours: number;
  total_storage_mb: number;
  generations_this_month: number;
  voices_this_month: number;
}

export interface UsageTimeSeriesPoint {
  date: string;
  count: number;
  type: string;
}

export interface UsageTimeSeriesResponse {
  data: UsageTimeSeriesPoint[];
  period: string;
}
